### ISSUE TITLE
Time-based file archival does not work correctly

### ISSUE THREAD

**Issue by kinshines** [NLog/NLog#2261]
**Type** :Bug

**NLog version**: 4.4.12

**Platform**:  .Net 4.5 

**Current NLog config** 

```xml
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="false"
      internalLogLevel="Off" internalLogFile="c:\temp\nlog-internal.log">

  <targets>
    <target xsi:type="File" name="file" encoding="utf-8" fileName="${basedir}\log\${logger}.log"
            layout="${time} ${message}" archiveEvery="Day" maxArchiveFiles="3" 
            archiveFileName="${basedir}\log\${logger}\${logger}_{#}.log" archiveNumbering="Date" />
  </targets>

  <rules>
    <logger name="*" minlevel="Debug" writeTo="file" />
  </rules>
</nlog>

```
**C# code** 
```c#
class Program
    {
        static void Main(string[] args)
        {
            LogInfo("module1", "message");
            LogInfo("module2", "message");
        }

        static void LogInfo(string module,string message)
        {
            Logger logger = LogManager.GetLogger(module);
            logger.Info(message);
        }
    }

```

- What is the current result?
I want to log message by module and every module will log one file in every day.
in date:2017-08-22,the program log two file: \log\module1.log and \log\module2.log,which is no problem
When it comes to date:2017-08-23,
the program moves the old file \log\module1.log to \log\module1\module1_**20170822**.log,
But it moves the old file \log\module2.log to \log\module2\module2_**20170823**.log
why the two module moved to archiveFileName with different date?
- What is the expected result?
Expected result is that the module2 also moved to \log\module2\module2_**20170822**.log
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
Internal log has nothing
- Please post full exception details (message, stacktrace, inner exceptions)
no more details
- Are there any work arrounds? 
no
- Is there a version in which it did worked? 
no
- Can you help us by writing an unit test?
if needed

**Comment by snakefoot** [comment]
The problem is the use of ${logger} in archiveFileName. This will be solved with #1993 (NLog ver. 4.5)

**Comment by 304NotModified** [comment]
fixed with https://github.com/NLog/NLog/pull/1993
